cmd/kops/toolbox_instance_selector.go

```diff
@@ -0,0 +1,487 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+
+	"github.com/aws/amazon-ec2-instance-selector/pkg/cli"
+	"github.com/aws/amazon-ec2-instance-selector/pkg/selector"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kops/cmd/kops/util"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/pkg/client/simple"
+	"k8s.io/kops/upup/pkg/fi/cloudup"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+// Filter Flag Constants
+const (
+	vcpus                  = "vcpus"
+	memory                 = "memory"
+	vcpusToMemoryRatio     = "vcpus-to-memory-ratio"
+	cpuArchitecture        = "cpu-architecture"
+	gpus                   = "gpus"
+	gpuMemoryTotal         = "gpu-memory-total"
+	placementGroupStrategy = "placement-group-strategy"
+	usageClass             = "usage-class"
+	enaSupport             = "ena-support"
+	burstSupport           = "burst-support"
+	availabilityZones      = "zones"
+	networkInterfaces      = "network-interfaces"
+	networkPerformance     = "network-performance"
+	allowList              = "allow-list"
+	denyList               = "deny-list"
+	maxResults             = "max-results"
+)
+
+// Aggregate Filter Flag Constants
+const (
+	instanceTypeBase = "instance-type-base"
+	flexible         = "flexible"
+)
+
+// Control Flag Constants
+const (
+	instanceGroupCount = "instance-group-count"
+	nodeCountMin       = "node-count-min"
+	nodeCountMax       = "node-count-max"
+	nodeVolumeSize     = "node-volume-size"
+	nodeSecurityGroups = "node-security-groups"
+	clusterAutoscaler  = "cluster-autoscaler"
+	usageClassSpot     = "spot"
+	usageClassOndemand = "on-demand"
+	igName             = "instance-group-name"
```

Comment:
I was getting a bit confused reading the code below, thinking the value of this variable was the name of the instance group, not the name of the flag. Perhaps we can name the variable to indicate that it is a flag.